.travis/dispatch.sh

```diff
@@ -3,24 +3,52 @@
 set -o errexit
 set -o nounset
 
-if [ "${SALT_NODE_ID}" = "test" ]; then
-    # Using .travis.yml to specify Python 3.5 to be preinstalled, just to check
-    printf "Using $(python3 --version) at $(which python3)\n"
+travis_fold_start () {
+    printf "travis_fold:start:$1\n"
+    printf "$2\n"
+}
 
-    # Run test suite separately for parallelism
-    ./test.py
-else
+travis_fold_end () {
+    printf "travis_fold:end:$1\n"
+}
+
+run_salt () {
+    travis_fold_start "salt.install.$1" 'Installing and configuring Salt'
     .travis/install_salt.sh -F -c .travis -- "${TRAVIS_OS_NAME}"
     .travis/setup_salt_roots.sh
+    travis_fold_end "salt.install.$1"
 
-    # For debugging, check the grains reported by the Travis builder
+    travis_fold_start "grains.items.$1" 'Printing Salt grains for debugging'
     sudo salt-call --id="${SALT_NODE_ID}" grains.items
+    travis_fold_end "grains.items.$1"
 
-    # Minimally validate YAML and Jinja at a basic level
+    travis_fold_start "state.show_highstate.$1" 'Performing basic YAML and Jinja validation'
     sudo salt-call --id="${SALT_NODE_ID}" --retcode-passthrough state.show_highstate
-    # Full on installation test
+    travis_fold_end "state.show_highstate.$1"
+
+    printf 'Running the full Salt highstate\n'
     sudo salt-call --id="${SALT_NODE_ID}" --retcode-passthrough --log-level=warning state.highstate
+}
+
+
+if [ "${SALT_NODE_ID}" = "test" ]; then
+    # Using .travis.yml to specify Python 3.5 to be preinstalled, just to check
+    printf "Using $(python3 --version) at $(which python3)\n"
+
+    # Run test suite separately for parallelism
+    ./test.py
+else
+    if [ "${SALT_FROM_SCRATCH}" = "true" ]; then
+        run_salt 'scratch'
+    else
+        git checkout master
```

Comment:
Travis doesn't seem to want you to do this; was it working in tests on your branch?

The approved way to use multiple branches seems to just be the matrix (https://docs.travis-ci.com/user/customizing-the-build/#Building-Specific-Branches), though I'm not sure if it'll meet this patch's needs.